pkg/arvo/sys/arvo.hoon

```diff
@@ -1241,7 +1251,7 @@
     ::
     ++  emit
       |=  pan=plan
-      this(run [pan run])
+      this(run (weld run pan ~))
```

Comment:
At the risk of slightly-more-complicated code, but without going to full-on finger trees, a commonly-used trick to get an amortized-kind-of-efficient deque with an SLL is to keep a backend list and a frontend list. decons off the frontend list, cons onto the backend list. When the frontend list is exhausted, reset it to the flop of the backend list and then null the backend list.

This means backend pushes are always O(1), and pops are amortized O(1).

Reply:
Oh yeah, I forgot this trick.  That's probably worth doing.